pkg/render/applicationlayer/applicationlayer_test.go

```diff
@@ -0,0 +1,146 @@
+			i++
+		}
+
+		ds := rtest.GetResource(resources, applicationlayer.L7LogCollectorDeamonsetName, common.CalicoNamespace, "apps", "v1", "DaemonSet").(*appsv1.DaemonSet)
+
+		// Check rendering of daemonset.
+		Expect(ds.Spec.Template.Spec.HostNetwork).To(BeTrue())
+		Expect(ds.Spec.Template.Spec.HostIPC).To(BeTrue())
+		Expect(ds.Spec.Template.Spec.DNSPolicy).To(Equal(corev1.DNSClusterFirstWithHostNet))
+		Expect(len(ds.Spec.Template.Spec.Volumes)).To(Equal(2))
+		Expect(len(ds.Spec.Template.Spec.Containers)).To(Equal(2))
+		Expect(len(ds.Spec.Template.Spec.Tolerations)).To(Equal(3))
+
+		// Ensure each volume rendered correctly.
+		dsVols := ds.Spec.Template.Spec.Volumes
+		expectedVolumes := []corev1.Volume{
+			{
+				Name: applicationlayer.EnvoyLogsVolumeName,
+				VolumeSource: corev1.VolumeSource{
+					EmptyDir: &corev1.EmptyDirVolumeSource{},
+				},
+			},
+			{
+				Name: applicationlayer.EnvoyConfigMapName,
+				VolumeSource: corev1.VolumeSource{
+					ConfigMap: &corev1.ConfigMapVolumeSource{
+						LocalObjectReference: corev1.LocalObjectReference{Name: applicationlayer.EnvoyConfigMapName},
+					},
+				},
+			},
+		}
+
+		for _, expected := range expectedVolumes {
+			Expect(dsVols).To(ContainElement(expected))
+		}
+
+		// Ensure that tolerations rendered correctly.
+		dsTolerations := ds.Spec.Template.Spec.Tolerations
+		expectedToleration := rmeta.TolerateAll
+		for _, expected := range expectedToleration {
+			Expect(dsTolerations).To(ContainElement(expected))
+		}
+
+		// Check proxy container rendering.
+		proxyContainer := ds.Spec.Template.Spec.Containers[0]
+
+		proxyEnvs := proxyContainer.Env
+		expectedProxyEnvs := []corev1.EnvVar{
+			{Name: "ENVOY_UID", Value: "0"},
+			{Name: "ENVOY_GID", Value: "0"},
+		}
+		for _, expected := range expectedProxyEnvs {
+			Expect(proxyEnvs).To(ContainElement(expected))
+		}
+
+		proxyVolMounts := proxyContainer.VolumeMounts
+		expectedProxyVolMounts := []corev1.VolumeMount{
+			{Name: applicationlayer.EnvoyConfigMapName, MountPath: "/etc/envoy"},
+			{Name: applicationlayer.EnvoyLogsVolumeName, MountPath: "/tmp/"},
+		}
+		for _, expected := range expectedProxyVolMounts {
+			Expect(proxyVolMounts).To(ContainElement(expected))
+		}
+
+		collectorContainer := ds.Spec.Template.Spec.Containers[1]
+
+		collectorEnvs := collectorContainer.Env
+		expectedCollectorEnvs := []corev1.EnvVar{
+			{Name: "LOG_LEVEL", Value: "Info"},
+			{Name: "FELIX_DIAL_TARGET", Value: "/var/run/felix/nodeagent/socket"},
+		}
+		for _, element := range expectedCollectorEnvs {
+			Expect(collectorEnvs).To(ContainElement(element))
+		}
+
+		collectorVolMounts := proxyContainer.VolumeMounts
+		expectedCollectorVolMounts := []corev1.VolumeMount{
+			{Name: applicationlayer.EnvoyLogsVolumeName, MountPath: "/tmp/"},
+		}
+		for _, expected := range expectedCollectorVolMounts {
+			Expect(collectorVolMounts).To(ContainElement(expected))
+		}
+	})
```

Comment:
I would like to see an additional test that sets the lcSpec to non-default values. You should only include checks for those specific values (no need to check everything that the above test does).